lib/matplotlib/_mathtext.py

```diff
@@ -2841,6 +2857,38 @@ def binom(self, s, loc, toks):
         return self._genfrac('(', ')', 0.0, self._MathStyle.TEXTSTYLE,
                              num, den)
 
+    def _genset(self, state, body, annotation, overunder):
+        thickness = state.font_output.get_underline_thickness(
+            state.font, state.fontsize, state.dpi)
+
+        body.shrink()
+
+        cbody = HCentered([body])
+        cannotation = HCentered([annotation])
+        width = max(cbody.width, cannotation.width)
+        cbody.hpack(width, 'exactly')
+        cannotation.hpack(width, 'exactly')
+
+        kern_vgap = thickness * 3
+        if overunder == "under":
+            vlist = Vlist([cannotation,                # annotation
+                           Kern(kern_vgap),            # space
```

Comment:
Can you (for my personal education) explain why this needs to be a Kern and not a Vbox?  Does that mean the other places that use Vboxes (e.g. `_genfrac`, `c_over_c`) should also be changed?

Reply:
We can understand `Kern` as a superset of `Vbox` and `Hbox` - `Vbox` is only used for vertical spaces, `Hbox` is only used for horizontal spaces.
`Kern` can be used in both - it is a more LaTeX-like keyword, so anybody who has an understanding of general LaTeX should feel at home looking at this.
On the other hand, `Vbox` should make more sense in vertical lists - since it is easy to understand what it would do.

Other than this, there is _no difference_ if we change `Kern` to a `Vbox` here, or other places (eg. `_genfrac`, `c_over_c`).

Reply:
Since we wouldn't want a _negative_ `kern_vgap`, I think it's better to stick to `VBox` here?

Reply:
Let's stick to Vbox for consistency with the other constructs (genfrac, c_over_c, etc.)?